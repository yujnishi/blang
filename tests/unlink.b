NULL 0;

main() {
    auto fname "unlink.tmp";

    close(2); dup(1); /* 2>&1 */

    if ( !fork() ) execl("/usr/bin/touch","touch",fname,NULL);
    wait();

    if ( !fork() ) execl("/bin/ls","ls",fname,NULL);
    wait();

    unlink(fname);

    if ( !fork() ) execl("/bin/ls","ls",fname,NULL);
    wait();
}

