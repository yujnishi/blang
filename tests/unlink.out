unlink.tmp
ls: cannot access unlink.tmp: No such file or directory
